src/engine/input/key-mapping/InputGroupList.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Godot;
+using Newtonsoft.Json;
+
+/// <summary>
+///   Used by OptionsMenu>Inputs>InputGroupContainer
+/// </summary>
+public class InputGroupList : VBoxContainer
+{
+    [Export]
+    public NodePath ConflictDialogPath;
+
+    internal static readonly PackedScene InputEventItemScene =
+        GD.Load<PackedScene>("res://src/engine/input/key-mapping/InputEventItem.tscn");
+
+    private static readonly Dictionary<string, List<InputEventWithModifiers>> DefaultControlScheme =
+        GetCurrentlyAppliedControlScheme();
+
+    private ConfirmationDialog conflictDialog;
+
+    private InputEventItem latestDialogCaller;
+    private InputEventItem latestDialogConflict;
+    private InputEventWithModifiers latestDialogNewEvent;
+
+    public InputGroupList()
+    {
+        Instance = this;
+    }
+
+    public delegate void ControlSchemeChangedDelegate(Dictionary<string, List<InputEventWithModifiers>> data);
+
+    public event ControlSchemeChangedDelegate OnControlSchemeChanged;
+
+    public static InputGroupList Instance { get; private set; }
```

Comment:
Instead of making this a static property, could it be made so that the child objects are passed callbacks etc. so that they don't need to access this like this?
Also this reference doesn't seem to be unset anywhere so the garbage collector can't dispose of this.

Reply:
It is now used by the pause menu, which means that I require this property.
The reference does not get unset per se, but it gets overritten when there is a new OptionsMenu getting loaded.
I think I have found another memory issue with circular references, which I will resolve.